fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -285,77 +330,78 @@ private void validate(Map<String, String> properties) throws DdlException {
             }
             if (!validColumnType(hudiColumn.schema(), column.getType())) {
                 throw new DdlException("Can not convert hudi column type [" + hudiColumn.schema().toString() + "] " +
-                        "to starrocks type [" + column.getPrimitiveType() + "], column name: " + column.getName());
+                        "to starrocks type [" + column.getPrimitiveType() + "], column name: " + column.getName()
+                        + ", starrocks type should be "
+                        + HiveMetaStoreTableUtils.convertHudiTableColumnType(hudiColumn.schema()).toSql());
             }
         }
+        return metastoreTable;
+    }
 
-        if (!copiedProps.isEmpty()) {
-            throw new DdlException("Unknown table properties: " + copiedProps);
+    public static Schema loadHudiSchema(org.apache.hadoop.hive.metastore.api.Table hiveTable) throws DdlException {
+        String hudiBasePath = hiveTable.getSd().getLocation();
+        Configuration conf = new Configuration();
+        HoodieTableMetaClient metaClient =
+                HoodieTableMetaClient.builder().setConf(conf).setBasePath(hudiBasePath).build();
+        TableSchemaResolver schemaUtil = new TableSchemaResolver(metaClient);
+        Schema hudiTable;
+        try {
+            hudiTable = HoodieAvroUtils.createHoodieWriteSchema(schemaUtil.getTableAvroSchema());
+        } catch (Exception e) {
+            throw new DdlException("Cannot get hudi table schema.");
         }
+        return hudiTable;
     }
 
-    private boolean validColumnType(Schema avroSchema, Type srType) {
+    public static boolean validColumnType(Schema avroSchema, Type srType) throws DdlException {
         Schema.Type columnType = avroSchema.getType();
         if (columnType == null) {
             return false;
         }
 
-        PrimitiveType primitiveType = srType.getPrimitiveType();
-        LogicalType logicalType = avroSchema.getLogicalType();
+        boolean valid = true;
+        Type convertType = HiveMetaStoreTableUtils.convertHudiTableColumnType(avroSchema);
+        // if convertType is ArrayType, so srType must be srType too
```

Comment:
It is obvious that, if an avroSchema is ArrayType, so the srType must be srType too.So, we can calculate the level of arrays in srType.
eg: level num of Array<Array<Array<int>>> is 3, and the primitive type of srType is int.